connect/runtime/src/main/java/org/apache/kafka/connect/runtime/ConnectMetrics.java

```diff
@@ -73,7 +74,7 @@ public ConnectMetrics(String workerId, WorkerConfig config, Time time) {
         reporters.add(new JmxReporter(JMX_PREFIX));
         this.metrics = new Metrics(metricConfig, reporters, time);
         LOG.debug("Registering Connect metrics with JMX for worker '{}'", workerId);
-        AppInfoParser.registerAppInfo(JMX_PREFIX, workerId, metrics);
+        AppInfoParser.registerAppInfo(JMX_PREFIX, Sanitizer.sanitize(workerId), metrics);
```

Comment:
@ewencp, would you agree that this sanitation might be better done inside the `AppInfoParser.registerAppInfo(...)` method itself?

Reply:
@rhauch I tried to follow the existing calls to the `registerAppInfo(...)` where the sanitation was done before the call. We should probably try to avoid sanitising twice. So if we put it into the method it self we will probably need to analyse the other calls to find out if the sanitation is done only because of the `registerAppInfo(...)` or whether it has also some other reasons.

Reply:
We do need to correct this, but I think the question is where. The problem is that many of the metric names should also be sanitized, but a recent [KIP-190 discussion on the mailing list](http://mail-archives.apache.org/mod_mbox/kafka-dev/201501.mbox/%3CCAOeJiJh6Vkkca85bWYgkeOZ8rC6%2BKDh7zzq8vMKECL_7PNExTA%40mail.gmail.com%3E) talked about doing this within the JMX Reporter since other reporters may not want sanitized names.

One option would be to merge this to at least fix the current code, but then to address the JMX Reporter sanitation separately (but still before 1.0). Thoughts, @ewencp?

Reply:
Yeah, we want to restrict sanitization to only locations where it is really needed, which in this case is for the "manually" created mbean for the app id in AppInfoParser. I can move the call there during commit while preserving your authorship, which I've verified works as expected. In this case it actually wouldn't matter much since that argument is only used for creating that one JMX metric, but I'd like it at the site so that during review of this other follow up item https://issues.apache.org/jira/browse/KAFKA-6011 we make sure the sanitization gets removed so other metrics reporters can report the correct value.

Thanks for the fix @scholzj!